modules/uploads/utils/types.ts

```diff
@@ -0,0 +1,81 @@
+import {
+	Files as PrismaFiles,
+	Upload as _PrismaUpload,
+} from "../_gen/prisma/default.d.ts";
+
+interface PrismaUpload extends Omit<_PrismaUpload, "deletedAt"> {
+	files: PrismaFiles[];
+}
+
+export interface Upload {
+	id: string;
+	userId: string | null;
+
+	bucket: string;
+	contentLength: string;
+
+	files: UploadFile[];
+
+	createdAt: string;
+	updatedAt: string;
+	completedAt: string | null;
+}
+
+export interface UploadFile {
+	path: string;
+	mime: string | null;
+	contentLength: string;
+}
+
+export interface PresignedUpload extends Omit<Upload, "files"> {
+	files: PresignedUploadFile[];
+}
+
+export interface PresignedUploadFile extends UploadFile {
+	presignedUrls: PresignedUrl[];
+}
+
+export interface PresignedUrl {
+	url: string;
+
+	partNumber: number;
+	contentLength: string;
+	offset: string;
+}
```

Comment:
The name `PresignedUrl` is a bit confusing here since it has more than just the url. maybe `PresignedUploadRequest`?